src/lib/index/shared/transport.js

```diff
@@ -39,20 +39,23 @@ const
 			transport = {
 				contextSchema: JSON.stringify(compiledContextSchema.toJSON()),
 				contextBuffer: compiledContextSchema.toBuffer(contextOrEmpty),
-				messageSchemaName: compiledMessageSchema.name,
+				messageSchema: JSON.stringify(compiledMessageSchema.toJSON()),
```

Comment:
Converting to and from JSON seems unusual. Perhaps an explanatory comment?

Reply:
Good point, I'll check.